gk/main.c

```diff
@@ -572,15 +719,30 @@ add_new_flow_from_policy(
 		return NULL;
 	}
 
-	/* Create a new flow entry. */
+	/*
+	 * We heuristically drop entries to alleviate memory pressure
+	 * when the table is full.
+	 */
+hash_add_flow:
 	ret = rte_hash_add_key_with_hash(
```

Comment:
Encapsulate into a new function the code that adds a new entry, and tries to drop an "old" entry when the table is full. This new function will be used here, `add_new_flow_from_policy()`, and in `process_pkts_front()`.

Reply:
It seems the failure handling functions for the two cases (i.e., ``add_new_flow_from_policy()``, and ``process_pkts_front()``) are not the same, it's non-trivial to write a general function to do so.

Reply:
If the new function just fails when it cannot add the entry even after looking for a candidate to drop, the caller can deal with the error in the specific way needed.